docs/software/other/mqtt.md

```diff
@@ -35,6 +36,44 @@ The payload is a raw protobuf. Looking at the MQTT traffic with a program like `
 	ShortFast	!937bed1c
 ```
 
+Packets of the types `TEXT_MESSAGE_APP`, `ENVIRONMENTAL_MEASUREMENT_APP`, `NODEINFO_APP` and `POSITION_APP` are encoded to JSON messages and sent to the `msh/1/json/CHANNELID/DEVICEID` topic. 
+
+An example of a received `NODEINFO_APP` message:
+```
+{
+   "id":452664778,
+   "channel":0,
+   "from":2130636288,
+   "payload":{
+      "hardware":10,
+      "id":"!7efeee00",
+      "longname":"base0",
+      "shortname":"BA0"
+   },
+   "sender":"!7efeee00",
+   "timestamp":1646832724,
+   "to":-1,
+   "type":"nodeinfo"
+}
+```
+
+If the message received contains valid JSON in the payload, the JSON is deserialized and added as a JSON object rather than a string containing the serialized JSON.
+
+**Sent messages** will be checked if the MQTT payload contains a valid JSON-encoded envelope:
+
+```
```

Comment:
Please specify language for code blocks: ```json

Reply:
Fixed in #6dd1de3.